src/services/ibex/client/index.ts

```diff
@@ -1,119 +1,112 @@
 // Ibex SDK API registry: https://dash.readme.com/api/v1/api-registry/cpd51bloegfhl2
 import IbexSDK, * as types from "./.api/apis/sing-in" // TODO: @sing-in@<uuid>
-import { IbexClientError, IbexAuthenticationError, IbexApiError } from "./errors"
+import { IbexAuthenticationError, IbexApiError } from "./errors"
 import { withAuth } from "./authentication";
-import { logRequest, logResponse } from "./errors/logger"
 import WebhookServer from "../webhook-server"
+import { addAttributesToCurrentSpan, wrapAsyncFunctionsToRunInSpan } from "@services/tracing"
 
-// This is a wrapper around the Ibex api that handles authentication
-class AuthenticatedIbexClient {
-    private static instance: AuthenticatedIbexClient | null = null;
-    private constructor() {}
-
-    static getInstance(): AuthenticatedIbexClient {
-        if (!AuthenticatedIbexClient.instance) {
-            AuthenticatedIbexClient.instance = new AuthenticatedIbexClient();
-        }
-        return AuthenticatedIbexClient.instance;
-    }
-
-    async getAccountTransactions(metadata: types.GMetadataParam): Promise<types.GResponse200 | IbexAuthenticationError | IbexApiError> {
-        logRequest("getAccountTransactions", metadata)
+// This is a wrapper around the Ibex api that adds tracing & authentication
+export default () => {
```

Comment:
any reason this is a function and not just an object?

Reply:
Yeah, I was trying to avoid doing this as well, but couldn't figure out to de-structure the class into the `wrapAsyncFunctionsToRunInSpan` function. If I remember correctly, I also tried to define it as an object, but couldn't figure that out either, so instead, I settled for this approach, which is similar to the how the rest of the code base is defined. 


IMO, this would be good to revisit at some point. But for now, I'd rather merge what works than spend more time trying to perfect